### ISSUE TITLE
LFRic PSyIR datatypes are really for kernel arguments currently

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#2425]
I've been trying to add some 'proper' Symbols into the code that creates the LFRic PSy layer. I tried to add a dof-map array but found that that gives me a rank 1 array whereas in the PSy layer, it is a rank 2 array - a 1D slice of it is passed down to a kernel.

    sym = self._symbol_table.find_or_create_tag(
                                map_name, fs=fs_name,
                                symbol_type=LFRicTypes("DofMapDataSymbol"),
                                dims=[Reference(ndf_symbol)])

Perhaps we need to prefix the existing type names with "Kernel" to distinguish them from those in the PSy layer? What do you think @rupertford?

**Comment by rupertford** [comment]
Yes, the LFRic-specific types and symbols are from the perspective of the kernel. I hadn't really thought about how we should name things through the different levels of PSyKAl but clearly some sort of convention that helps us relate something that is passed from the algorithm layer, or extracted from infrastructure with something that is passed in to the kernel and/or used in the kernel could be useful to help make the code more readable (and help us manage code generation).

I don't know whether it makes sense to mark everything that is used in a kernel with a particular prefix. Perhaps we should think through some of the actual cases first to see what makes sense e.g. a scalar, a field, a field vector, dofmaps, etc.?
